CHANGES.md

```diff
@@ -1,7 +1,7 @@
 # Changes
 
 ## Unreleased
-
+- Fixed: Calling `map(..).flatten()` not allowed on an `Option` from in `rust >= 1.59`
```

Comment:
It is allowed. The error you got is from `clippy`, which is a code linter for style problems and was recently added to the CI.

It's not worth adding this to the changelog, since it's only an internal change.